src/stdlib_linalg.fypp

```diff
@@ -108,4 +201,200 @@ contains
+           return !nonsquare matrices cannot be symmetric
+        end if
+        n = size(A,1) !symmetric dimension of A
+        do j = 1, n !loop over all columns
+            do i = 1, j-1 !loop over all rows above diagonal
+                if (A(i,j) .ne. A(j,i)) then
+                  res = .false.
+                  return
+                end if
+            end do
+        end do
+        res = .true. !otherwise A is symmetric
+      end function is_symmetric_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+      pure function is_skew_symmetric_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        integer :: n, i, j
+        if (.not. is_square(A)) then
+           res = .false.
+           return !nonsquare matrices cannot be skew-symmetric
+        end if
+        n = size(A,1) !symmetric dimension of A
+        do j = 1, n !loop over all columns
+            do i = 1, j !loop over all rows above diagonal (and diagonal)
+                if (A(i,j) .ne. -A(j,i)) then
+                  res = .false.
+                  return
+                end if
+            end do
+        end do
+        res = .true. !otherwise A is skew-symmetric
+      end function is_skew_symmetric_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in (REAL_KINDS_TYPES + INT_KINDS_TYPES)
+      pure function is_hermitian_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        if (.not. is_square(A)) then
+           res = .false.
+           return !nonsquare matrices cannot be Hermitian
+        end if
+        res = is_symmetric(A) !real symmetric matrices are Hermitian
+      end function is_hermitian_${t1[0]}$${k1}$
+    #:endfor
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+      pure function is_hermitian_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        integer :: n, i, j
+        if (.not. is_square(A)) then
+           res = .false.
+           return !nonsquare matrices cannot be Hermitian
+        end if
+        n = size(A,1) !symmetric dimension of A
+        do j = 1, n !loop over all columns
+            do i = 1, j !loop over all rows above diagonal (and diagonal)
+                if (A(i,j) .ne. conjg(A(j,i))) then
+                  res = .false.
+                  return
+                end if
+            end do
+        end do
+        res = .true. !otherwise A is Hermitian
+      end function is_hermitian_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+      function is_triangular_${t1[0]}$${k1}$(A,uplo) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        character, intent(in) :: uplo
+        logical :: res
+        ${t1}$ :: zero
+        integer :: m, n, o, i, j
+        zero = 0 !zero of relevant type
+        m = size(A,1)
+        n = size(A,2)
+        if ((uplo .eq. 'u') .or. (uplo .eq. 'U')) then !check for upper triangularity
```

Comment:
```suggestion
       if (uplo == 'u' .or. uplo == 'U') then !check for upper triangularity
```

Or even

```suggestion
       if (to_lower(uplo) == 'u') then !check for upper triangularity
```

where `to_lower` is imported from `stdlib_ascii`.

Reply:
Personally, I'm kind of in favor of keeping the original here as it 1) keeps the independence of modules and 2) is less expensive than the call to `to_lower`. Granted, such matrix checks are typically not a performance bottleneck.

Reply:
@milancurcic, do you agree to leave the character checking as is?

Reply:
Yes, not using `to_lower` is fine, though I recommend using the modern relational operators here as well (this may have been updated already)